packages/dataviews/tsconfig.json

```
@@ -0,0 +1,20 @@
+{
+	"$schema": "https://json.schemastore.org/tsconfig.json",
+	"extends": "../../tsconfig.base.json",
+	"compilerOptions": {
+		"rootDir": "src",
+		"declarationDir": "build-types"
+	},
+	"references": [
+		{ "path": "../a11y" },
+		{ "path": "../components" },
+		{ "path": "../compose" },
+		{ "path": "../element" },
+		{ "path": "../i18n" },
+		{ "path": "../icons" },
+		{ "path": "../keycodes" },
+		{ "path": "../primitives" },
+		{ "path": "../private-apis" }
+	],
+	"include": [ "src/**/*.ts", "src/**/*.tsx" ]
```

This seems like the preferred way of including src but not typechecking `*.js` files:

```suggestion
	"compilerOptions": {
		"rootDir": "src",
		"declarationDir": "build-types",
		"checkJs": false
	},
	"references": [
		{ "path": "../a11y" },
		{ "path": "../components" },
		{ "path": "../compose" },
		{ "path": "../element" },
		{ "path": "../i18n" },
		{ "path": "../icons" },
		{ "path": "../keycodes" },
		{ "path": "../primitives" },
		{ "path": "../private-apis" }
	],
	"include": [ "src" ]
```

I think there's an ongoing problem with react and ariakit types that I expect to be fixed by https://github.com/WordPress/gutenberg/pull/60796. Sometimes [I get errors like this](https://github.com/WordPress/gutenberg/pull/60999#issuecomment-2072342629):

> node_modules/@ariakit/react-core/cjs/tooltip/tooltip-provider.d.ts:14:100 - error TS2694: Namespace '"…/node_modules/@types/react/jsx-runtime"' has no exported member 'JSX'.
>
> 14 export declare function TooltipProvider(props?: TooltipProviderProps): import("react/jsx-runtime").JSX.Element;

We can work around that in this PR if its a problem by adding `"skipLibCheck": true` to the `compilerOptions`.